### ISSUE TITLE
Use the same Mongoose version for Connection and Schema

### ISSUE THREAD

**Issue by bra1n** [donpark/session-mongoose#24]
Hi,

I just noticed that you put "mongoose: *" into the package.json, which will lead to getting unstable versions for Mongoose on an "npm update". This would be fine, by itself, but problems occur when you create the Schema instance with a differnet Mongoose version than the one used for the database connection.
Here's where it breaks:
- [you create the Schema with your own require'd version of Mongoose](https://github.com/donpark/session-mongoose/blob/master/index.coffee#L29), usually the latest unstable 
- [you then, however, try to register this Schema to a Mongoose connection from the main app](https://github.com/donpark/session-mongoose/blob/master/index.coffee#L45), which is probably a different, stable version of Mongoose.
  And that's where things break:

```
/node_modules/mongoose/lib/schema.js:362
    throw new TypeError('Undefined type at `' + path +
          ^
TypeError: Undefined type at `paths.sid`
  Did you try nesting Schemas? You can only nest using refs or arrays.
    at Function.Schema.interpretAsType (/node_modules/mongoose/lib/schema.js:362:11)
    at Schema.path (/node_modules/mongoose/lib/schema.js:305:29)
    at Schema.add (/node_modules/mongoose/lib/schema.js:217:12)
    at Schema.add (/node_modules/mongoose/lib/schema.js:212:14)
    at new Schema (/node_modules/mongoose/lib/schema.js:73:10)
    at NativeConnection.Connection.model (/node_modules/mongoose/lib/connection.js:584:14)
    at new SessionStore (/node_modules/session-mongoose/index.js:95:35)
    at Object.<anonymous> (/app.js:51:10)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
```

Suggested solution: if a `connection` is passed through the options parameter, you should use the Schema class from that connection, rather than the one from your own Mongoose version, to avoid conflicts.

Cheers,
Steffen

**Comment by donpark** [comment]
Hi @bra1n.

As suggested, version 0.5 uses `connection.Schema` if a connection is provided via options.

Let me know if you see any problems with this fix. Thx.

**Comment by bra1n** [comment]
Hey, thanks for the quick response! Did your changes work for you? Because it seems that Mongoose doesn't have the `Schema` reference bound to `connection`, at least that's what it tells me:

```
/node_modules/session-mongoose/index.js:62
              type: Schema.Types.Mixed,
                          ^
TypeError: Cannot read property 'Types' of undefined
    at new SessionStore (/node_modules/session-mongoose/index.js:62:27)
    at Object.<anonymous> (/app.js:51:10)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:906:3
```

**Comment by donpark** [comment]
I tested using REPL with following commands:

```
var Mongoose = require('mongoose');
var mongoose = new Mongoose.Mongoose();
var connection = mongoose.connect("mongodb://localhost/sessions");
console.dir(connection.Schema);
```

which outputs:

```
coffee> console.dir connection.Schema
{ [Function: Schema]
  reserved: 
   { _posts: 1,
     _pres: 1,
     _events: 1,
     emit: 1,
     toObject: 1,
     collection: 1,
     modelName: 1,
     options: 1,
     schema: 1,
     errors: 1,
     isNew: 1,
     init: 1,
     db: 1,
     on: 1 },
  interpretAsType: [Function],
  Types: 
   { String: [Function: SchemaString],
     Number: [Function: SchemaNumber],
     Boolean: { [Function: SchemaBoolean] '$conditionalHandlers': [Object] },
     DocumentArray: [Function: DocumentArray],
     Array: [Function: SchemaArray],
     Buffer: [Function: SchemaBuffer],
     Date: [Function: SchemaDate],
     ObjectId: [Function: ObjectId],
     Mixed: [Function: Mixed],
     Oid: [Function: ObjectId],
     Object: [Function: Mixed],
     Bool: { [Function: SchemaBoolean] '$conditionalHandlers': [Object] } },
  ObjectId: [Function: ObjectId] }
```

Maybe Schmea is there only after connection is made.

If you have more reliable way to retrieve a connection's Schema, let me know.

**Comment by donpark** [comment]
Found the problem which was a typo. Instead of:

```
Schema = connection.Schema
```

I had:

```
Schema = connection.schema
```

which results in `undefined`.

Added check to make sure Schema is never undefined. Fix is in version 0.5.1.

**Comment by bra1n** [comment]
Ran a quick test, seems to work now! :+1: 
Thank you very much. :)
